nixos/modules/services/mail/listmonk.nix

```
@@ -0,0 +1,212 @@
+              default = [];
+              description = lib.mdDoc "Administrator emails for system notifications";
+            };
+
+            "privacy.exportable" = mkOption {
+              type = listOf str;
+              default = ["profile" "subscriptions" "campaign_views" "link_clicks"];
+              description = lib.mdDoc "List of fields which can be exported through an automatic export request";
+            };
+
+            "privacy.domain_blocklist" = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "E-mail addresses with these domains are disallowed from subscribing.";
+            };
+
+            smtp = mkOption {
+              type = listOf (submodule {
+                freeformType = with types; attrsOf (oneOf [ str int bool ]);
+
+                options = {
+                  enabled = mkEnableOption "this SMTP server for listmonk";
+                  host = mkOption {
+                    type = types.str;
+                    description = lib.mdDoc "Hostname for the SMTP server";
+                  };
+                  port = mkOption {
+                    type = types.port;
+                    description = lib.mdDoc "Port for the SMTP server";
+                  };
+                  max_conns = mkOption {
+                    type = types.int;
+                    description = lib.mdDoc "Maximum number of simultaneous connections, defaults to 1";
+                    default = 1;
+                  };
+                  tls_type = mkOption {
+                    type = types.enum [ "none" "STARTTLS" "TLS" ];
+                    description = lib.mdDoc "Type of TLS authentication with the SMTP server";
+                  };
+                };
+              });
+
+              description = lib.mdDoc "List of outgoing SMTP servers";
+            };
+
+            # TODO: refine this type based on the smtp one.
+            "bounce.mailboxes" = mkOption {
+              type = listOf (submodule {
+                freeformType = with types; oneOf [ str int bool ];
+              });
+              default = [];
+              description = lib.mdDoc "List of bounce mailboxes";
+            };
+
+            messengers = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "List of messengers, see: <https://github.com/knadh/listmonk/blob/master/models/settings.go#L64-L74> for options.";
+            };
+          };
+        });
+
+        description = lib.mdDoc "Dynamic settings in the PostgreSQL database, set by a SQL script, see <https://github.com/knadh/listmonk/blob/master/schema.sql#L177-L230> for details.";
+      };
+      secretFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = lib.mdDoc "Secret file loaded as an environment file, read <https://listmonk.app/docs/configuration/#environment-variables> for details.";
```

```suggestion
        description = lib.mdDoc "A file containing secrets as environment variables. See <https://listmonk.app/docs/configuration/#environment-variables> for details on supported values.";
```

We should avoid any terminology that implies we are using `EnvironmentFile` so we don't propagate any misunderstanding that `EnvironmentFile` is either safe or suitable for secrets.

Unfortunately I can't recommend this PR adopt `LoadCredential` because currently it doesn't support `ExecStartPre`.

See https://github.com/NixOS/nixpkgs/pull/150771#discussion_r901022732 for previous comments I have made on this topic.

---

Yes, I do agree.